crates/core/tedge_mapper/src/az/converter.rs

```diff
@@ -44,13 +47,29 @@ impl Converter for AzureConverter {
     }
 
     async fn try_convert(&mut self, input: &Message) -> Result<Vec<Message>, Self::Error> {
+        let maybe_child_id = get_child_id_from_measurement_topic(&input.topic.name)?;
+
+        let az_out_topic = match maybe_child_id {
+            Some(child_id) => Topic::new_unchecked(&format!("az/messages/events/$.sub={child_id}")),
+            None => self.mapper_config.out_topic.clone(),
+        };
+
         self.size_threshold.validate(input)?;
         let default_timestamp = self.add_timestamp.then(|| self.clock.now());
         let mut serializer = ThinEdgeJsonSerializer::new_with_timestamp(default_timestamp);
         tedge_api::parser::parse_str(input.payload_str()?, &mut serializer)?;
 
         let payload = serializer.into_string()?;
-        Ok(vec![(Message::new(&self.mapper_config.out_topic, payload))])
+        Ok(vec![(Message::new(&az_out_topic, payload))])
+    }
+}
+
+pub fn get_child_id_from_measurement_topic(topic: &str) -> Result<Option<String>, ConversionError> {
```

Comment:
This same function exists in `c8y::converter.rs`. Both could be moved to a common module like `core::converter.rs` and be reused here and in the c8y converter.

Reply:
This should be part of the `tedge_api`. It's okay to keep the current PR as is.